address-controller/src/main/resources/swagger.json

```diff
@@ -0,0 +1,640 @@
+{
+    "swagger": "2.0",
+    "info": {
+        "description": "This is the EnMasse API specification.",
+        "version": "1.0.0",
+        "title": "EnMasse API"
+    },
+    "basePath": "/v1",
+    "tags": [
+        {
+            "name": "addressspaces",
+            "description": "Operate on AddressSpaces",
+            "externalDocs": {
+                "description": "Find out more",
+                "url": "http://enmasse.io"
+            }
+        },
+        {
+            "name": "addresses",
+            "description": "Operating on Addresses within an AddressSpace.",
+            "externalDocs": {
+                "description": "Find out more",
+                "url": "http://enmasse.io"
+            }
+        }
+    ],
+    "schemes": [
+        "http"
+    ],
+    "paths": {
+        "/addressspaces": {
+            "get": {
+                "tags": [
+                    "addressspaces"
+                ],
+                "summary": "Get list of all AddressSpaces",
+                "description": "Returns a list of all AddressSpaces in the system",
+                "operationId": "getAddressSpaceList",
+                "produces": [
+                    "application/json"
+                ],
+                "responses": {
+                    "200": {
+                        "description": "successful operation",
+                        "schema": {
+                            "$ref": "#/definitions/AddressSpaceList"
+                        }
+                    }
+                }
+            },
+            "post": {
+                "tags": [
+                    "addressspaces"
+                ],
+                "summary": "Create a new AddressSpace",
+                "description": "Creates a new AddressSpace in which you can then create Addresses.",
+                "operationId": "createAddressSpace",
+                "consumes": [
+                    "application/json"
+                ],
+                "produces": [
+                    "application/json"
+                ],
+                "parameters": [
+                    {
+                        "in": "body",
+                        "name": "body",
+                        "description": "AddressSpace object",
+                        "required": true,
+                        "schema": {
+                            "$ref": "#/definitions/AddressSpace"
+                        }
+                    }
+                ],
+                "responses": {
+                    "405": {
+                        "description": "Invalid input"
+                    }
```

Comment:
You probably want to return a 201 on successful creation, or a 202 if it's async.  Probably also want to return the created resource URI.

Reply:
It currently returns 200. I've added 200 to the spec now. Thanks